tests/integration/services/beneficiary.test.ts

```diff
@@ -323,4 +328,131 @@ describe('beneficiary service', () => {
             expect(result.length).to.be.equal(0);
         });
     });
+
+    describe('beneficiary activity', () => {
+        before(async () => {
+            const randomWallet = ethers.Wallet.createRandom();
+
+            const tx = randomTx();
+            const tx2 = randomTx();
+
+            await BeneficiaryService.add(
+                users[16].address,
+                users[0].address,
+                communities[0].publicId,
+                tx,
+                new Date('2021-01-01')
+            );
+
+            await BeneficiaryService.addTransaction({
+                beneficiary: users[16].address,
+                withAddress: await randomWallet.getAddress(),
+                amount: '25',
+                isFromBeneficiary: true,
+                tx,
+                date: new Date(), // date only
+            });
+
+            await BeneficiaryService.addTransaction({
+                beneficiary: users[16].address,
+                withAddress: await randomWallet.getAddress(),
+                amount: '50',
+                isFromBeneficiary: false,
+                tx: tx2,
+                date: new Date(), // date only
+            });
+
+            await ClaimsService.add({
+                address: users[16].address,
+                communityId: communities[0].id,
+                amount: '15',
+                tx,
+                txAt: new Date('2021-01-02'),
```

Comment:
I'm confused with what you've tried to do here :thinking: 
If you need time travel see this https://github.com/impactMarket/api/blob/master/tests/integration/jobs/cron/calcuateGlobalMetrics.test.ts#L84
Also, you can use factories if it helps.

Reply:
in this case, in each transaction, I forced a date so that I could test the ordering in the listing, from most recent to oldest.